tests/test_basic_auth.py

```diff
@@ -0,0 +1,143 @@
+import os
+import socket
+from urllib import parse as url_parse
+
+import requests
+import pytest
+import requests_mock
+
+from domino import Domino, bearer_auth
+
+
+def domino_is_reachable(url=os.getenv('DOMINO_HOST_KEY_NAME'), port='443'):
+    """
+    Confirm that a deployment is acessible for tests that require it.
+    """
+    if url is None:
+        return False
+
+    try:
+        fqdn = url_parse.urlsplit(url).netloc
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.connect((fqdn, int(port)))
+        is_reachable = True
+    except socket.gaierror:
+        print(f"{fqdn}:{port} is not reachable")
+        is_reachable = False
+    finally:
+        s.close()
+
+    return is_reachable
+
+
+@pytest.fixture
+def mock_domino_version_response():
+    """
+    Simulate a valid repsonse to an authenticated /version GET request.
+    """
+    version_info = {
+        "buildId": "12345",
+        "buildUrl": "https://some.server.com/domino/12345",
+        "commitId": "123deadbeef456",
+        "commitUrl": "https://some.repo.com/domino/commit/123deadbeef456",
+        "timestamp": "2021-06-14T16:50:02Z",
+        "version": "9.9.9"
+    }
+
+    with requests_mock.Mocker() as mock_endpoint:
+        mock_endpoint.get("http://mock.domino.com/version", json=version_info, status_code=200)
+        yield
+
+
+@pytest.fixture
+def dummy_token_file(tmpdir):
+    """
+    Simulate a token file.
+    """
+    token_file = tmpdir.join("dummy_token_file.txt")
+    token_file.write('top_secret_auth_token')
+    return token_file
+
+
+@pytest.fixture
+def real_token_file(tmpdir):
+    """
+    Derive a real authentication token (assumes that certain packages are available).
+    If any exception is thrown, returning None can signal tests to XFAIL.
+    """
+    try:
+        from domino_pytk.authentication import KeycloakToken
+        _url = os.getenv('DOMINO_HOST_KEY_NAME')
+        _user = os.getenv('DOMINO_USER')
+        _pw = os.getenv('DOMINO_PASSWORD')
+        keycloak_token = KeycloakToken(url=_url, username=_user, password=_pw)
+
+        token_file = tmpdir.join("tmp_token_file.txt")
+        token_file.write(keycloak_token)
+        os.environ['DOMINO_TOKEN_FILE_KEY_NAME'] = str(token_file)
+        return str(token_file)
+    except Exception as e:
+        print(e)
+        return None
+
+
+@pytest.mark.usefixtures("mock_domino_version_response")
+def test_object_creation_with_api_key():
+    """
+    Confirm that the expected auth type is used when using an api key.
+    """
+    dummy_host = "http://mock.domino.com"
```

Comment:
Changed to use our actual domain. I realize it doesn't matter for real, but still...
```suggestion
    dummy_host = "http://mock.dominodatalab.com"
```